userspace/libsinsp/sinsp.cpp

```diff
@@ -2795,6 +2820,27 @@ libsinsp::event_processor::build_threadinfo(sinsp* inspector)
 	return new sinsp_threadinfo(inspector);
 }
 
+void sinsp::handle_async_event(std::unique_ptr<sinsp_evt> evt)
+{
+	// see comments in handle_plugin_async_event
+	if(!is_capture())
```

Comment:
I would remove or convert this `if` into `ASSERT()`, if we arrive here and we are in capture mode there is definitely something wrong, WDYT?